src/SampleWebApp/Controllers/BooksMvcController.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+
+using SampleWebApp.Data;
+using SampleWebApp.Models;
+
+namespace SampleWebApp.Controllers
+{
+    public class BooksMvcController : Controller
+    {
+        private BookDataContext _context;
+
+        public BooksMvcController(BookDataContext context)
+        {
+            _context = context;    
+        }
+
+        // GET: BooksMvc
+        public IActionResult Index()
+        {
+            var bookDataContext = _context.Book.Include(b => b.Author);
+            return View(bookDataContext.ToList());
+        }
+
+        // GET: BooksMvc/Details/5
+        public IActionResult Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Book book = _context.Book.SingleOrDefault(m => m.BookID == id);
+            if (book == null)
+            {
+                return NotFound();
+            }
+
+            return View(book);
+        }
+
+        // GET: BooksMvc/Create
+        public IActionResult Create()
+        {
+            ViewData["AuthorID"] = new SelectList(_context.Set<Author>(), "AuthorID", "Author");
```

Comment:
Why are we using `ViewData`? I know the select helpers have magic logic in them to use it but I personally **hate** it. Can we please make it scaffold explicit stuff instead?

Also, use `nameof(Author.AuthorID)`, etc. instead of strings for property names.